lib/shared/src/sourcegraph-api/graphql/client.ts

```diff
@@ -1311,29 +1316,44 @@ export class SourcegraphGraphQLAPIClient {
         includeViewerDrafts?: boolean
         builtinOnly?: boolean
     }): Promise<Prompt[]> {
-        const hasIncludeViewerDraftsArg = await this.isValidSiteVersion({
-            minimumVersion: '5.9.0',
-        })
-        const hasPromptTagsField = await this.isValidSiteVersion({
-            minimumVersion: '5.11.0',
-            insider: true,
-        })
+        const [hasIncludeViewerDraftsArg, hasPromptTagsField, hasOrderByRelevance] = await Promise.all([
+            this.isValidSiteVersion({
+                minimumVersion: '5.9.0',
+            }),
+            this.isValidSiteVersion({
+                minimumVersion: '5.11.0',
+            }),
+            this.isValidSiteVersion({
+                minimumVersion: '6.3.1',
+            }),
+        ])
+
+        const gqlQuery = hasOrderByRelevance
+            ? PROMPTS_QUERY
+            : hasIncludeViewerDraftsArg
+              ? LEGACY_PROMPTS_QUERY_6_3
+              : LEGACY_PROMPTS_QUERY_5_8
+
+        const input = {
+            query,
+            first: first ?? 100,
+            recommendedOnly: recommendedOnly,
+            tags: hasPromptTagsField ? tags : undefined,
+            owner,
+            includeViewerDrafts: includeViewerDrafts ?? true,
+            builtinOnly,
+            orderBy: hasOrderByRelevance ? PromptsOrderBy.PROMPT_RELEVANCE : undefined,
+            orderByMultiple: hasOrderByRelevance
+                ? undefined
+                : orderByMultiple || [
+                      PromptsOrderBy.PROMPT_RECOMMENDED,
+                      PromptsOrderBy.PROMPT_UPDATED_AT,
+                  ],
+        }
```

Comment:
```suggestion
        const input = {
            query,
            first,
            recommendedOnly,
            owner,
            includeViewerDrafts,
            builtinOnly,
            ...(hasPromptTagsField && tags ? { tags } : {}),
            ...(hasOrderByRelevance
                ? { orderBy: PromptsOrderBy.PROMPT_RELEVANCE }
                : {
                      orderByMultiple: orderByMultiple || [
                          PromptsOrderBy.PROMPT_RECOMMENDED,
                          PromptsOrderBy.PROMPT_UPDATED_AT,
                      ],
                  }),
        }
```
Use object spread instead?